app/server/methods/data.js

```diff
@@ -1,10 +1,73 @@
+const async = require("async");
+
+async function _exportDataToConcernedColection(collectionName, data) {
+  if (!Array.isArray(data)) throw "Data not in required format";
+
+  console.log("Calling data for coll: ", collectionName);
+
+  switch (collectionName) {
```

Comment:
Instead of using a switch, use a dict:

```js
// collections are currently global variables,
// so no need to import
const collections = {
  activities: Activities,
  residents: Residents,
  ...
}
```

Then, you can just do:

```js
// async should not be needed, as Meteor will do this for us
// just use the collection .insert() method directly, no indirection
collections[collectionName].insert([ array of documents ]);
```